integration-tests/utils/integration-utils.js

```diff
@@ -428,8 +400,35 @@ const withPageLogging = fn => async options => {
 
 const navOptionNetworkIdle = (timeout = 60 * 1000) => ({ waitUntil: ['networkidle0'], timeout })
 
-const gotoPage = (page, url) => {
-  return page.goto(url, navOptionNetworkIdle())
+const gotoPage = async (page, url) => {
+  const retryOptions = {
+    factor: 1,
+    onFailedAttempt: error => {
+      console.error(
+        `Loading url attempt ${error.attemptNumber} failed. There are ${error.retriesLeft} retries left.`
+      )
+    },
+    retries: 2
+  }
+
+  const load = async url => {
+    try {
+      const httpResponse = await page.goto(url, navOptionNetworkIdle())
+      if (httpResponse && !(httpResponse.ok() || httpResponse.status() === 304)) {
+        throw new Error(`Error loading URL: ${url}. Http response status: ${httpResponse.statusText()}`)
+      }
```

Comment:
`page.goto` returns "the main resource response". The PR description mentions 404 errors from requests for JS files, not the initial request for the page. I don't think this would detect those types of errors.

https://pptr.dev/15.3.0/api/puppeteer.page.goto

Reply:
I'm moving code that handles sign-in issue from `signInToTerra` to `gotoPage` because few tests don't use `signInToTerra` to load the first page and I'm not duplicate code.

Puppeteer `goto(testUrl)` does not know one failed request out of many redirect requests (in this case, triggered by loading of testUrl).

`goto(testUrl)` returns either `null`, `ok` or `304` code (so far that's what I've seen). 

L420 fails because `Loading Terra ...` spinner spins forever, due to one request out many failed with 404.

Reply:
Ooh, I see. The 404 response from the chunk request is not detected here. Failing to load a chunk causes `page.waitForXPath('//*[contains(normalize-space(.),"Loading Terra")]', { hidden: true })` to timeout and then the page is retried.